core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -700,7 +700,7 @@ public void processPendingDeletes(Index index, IndexSettings indexSettings, Time
                             sleepTime = Math.min(maxSleepTimeMs, sleepTime * 2); // increase the sleep time gradually
                             logger.debug("{} schedule pending delete retry after {} ms", index, sleepTime);
                         } catch (InterruptedException e) {
-                            Thread.interrupted();
+                            Thread.currentThread().interrupt();
```

Comment:
Agreed. However, as I was trying to explain in my comment, we're dealing with interruption inconsistently in this method. When we're interrupted while we're stuck in `InternalShardLock#acquire()` we throw an exception and will *not* notify the master (i.e. we will not call the second method `TransportService#sendRequest()` from `IndicesService#lockIndexAndAck()`) but if we are interrupted while we are in a timeout we will still notify the master (as we just return). If that's ok, then I'm fine with that. :)